src/plugins/nest.js

```diff
@@ -0,0 +1,294 @@
+'use strict'
+
+function createWrapCreate (tracer, config) {
+  return function wrapCreate (create) {
+    return function createWithTrace () {
+      const instance = arguments[0]
+      const methodName = arguments[2]
+
+      const handler = create.apply(this, arguments)
+      return function (req, res, next) {
+        let opName = 'nest.request'
+        const scope = tracer.scope()
+        const childOf = scope.active()
+        const span = tracer.startSpan(opName, {
+          childOf,
+          tags: {
+            'component': 'nest',
+            'nest.request.method': req.method,
+            'nest.request.path': req.route.path
+          }
+        })
+
+        if (instance.constructor && instance.constructor.name) {
+          opName = instance.constructor.name
+        }
+        span.setTag('resource.name', `${opName}.${methodName}`)
+        return scope.activate(span, () => {
+          try {
+            return handler.apply(this, arguments)
+          } catch (e) {
+            throw addError(span, e)
+          } finally {
+            span.finish()
+          }
+        })
+      }
+    }
+  }
+}
+
+function createWrapNestFactoryCreate (tracer, config) {
+  return function wrapCreate (create) {
+    return function createWithTrace (module, serverOrOptions, options) {
+      const scope = tracer.scope()
+      const childOf = scope.active()
+      const span = tracer.startSpan('nest.factory.create', {
+        childOf,
+        tags: {
+          'component': 'nest',
+          'nest.module': module.name
+        }
+      })
+
+      return scope.activate(span, () => {
+        try {
+          return create.apply(this, arguments)
+        } catch (e) {
+          throw addError(span, e)
+        } finally {
+          span.finish()
+        }
+      })
+    }
+  }
+}
+
+function createWrapIntercept (tracer, config) {
+  return function wrapIntercept (intercept) {
+    return function interceptWithTrace () {
+      const interceptors = arguments[0]
+      const instance = arguments[2]
+      const request = arguments[1][0]
+      const callbackFnName = arguments[3].name
+
+      const opName = 'nest.interceptor.intercept'
+      const scope = tracer.scope()
+      const childOf = scope.active()
+      const span = tracer.startSpan(opName, {
+        childOf,
+        tags: {
+          'component': 'nest'
+        }
+      })
+      if (callbackFnName) {
```

Comment:
Would prefer you update the signature and check for existence of name on callback instead of spreading this functionality over multiple, disjointed lines.